File: jcv-core/src/main/java/com/ekino/oss/jcv/core/validator/ValueTemplateIdValidator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator;

import com.ekino.oss.jcv.core.JsonContextMatcher;
import com.ekino.oss.jcv.core.JsonValidator;

abstract class ValueTemplateIdValidator<T> implements JsonValidator<T> {

    private final ValidatorIdInValueMatcher matcher;

    public ValueTemplateIdValidator(String validatorId) {
        this.matcher = new ValidatorIdInValueMatcher(validatorId);
    }

    @Override
    public JsonContextMatcher getContextMatcher() {
        return matcher;
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/validator/ValueTemplateIdValidator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.validator

import com.ekino.oss.jcv.core.JsonContextMatcher
import com.ekino.oss.jcv.core.JsonValidator

abstract class ValueTemplateIdValidator<T>(validatorId: String) : JsonValidator<T> {

  private val matcher: ValidatorIdInValueMatcher = ValidatorIdInValueMatcher(validatorId)

  override fun getContextMatcher(): JsonContextMatcher = matcher
}
